index.js

```diff
@@ -112,7 +140,10 @@ Client.prototype.flush = function() {
   if (this.buffer.length > 0) {
     this.send(this.buffer);
     this.buffer = '';
+    this.lastFlushed = new Date();
   }
+
+  this.flushScheduled = false;
```

Comment:
Looking at this, setInterval is probably more idiomatic than trying to do this flush logic. The dev readability will be more than worth the probably-negligible performance issues.